modules/fedimint-mint-client/src/lib.rs

```diff
@@ -110,11 +119,25 @@ impl OOBNotes {
         ])
     }
 
+    pub fn new_with_invite(notes: TieredMulti<SpendableNote>, invite: InviteCode) -> Self {
+        Self(vec![
+            // FIXME: once we can break compatibility with 0.2 we can remove the prefix in case an
+            // invite is present
+            OOBNotesData::FederationIdPrefix(invite.federation_id().to_prefix()),
+            OOBNotesData::Notes(notes),
+            OOBNotesData::Invite {
+                peer_apis: vec![(invite.peer(), invite.url())],
+                federation_id: invite.federation_id(),
+            },
+        ])
+    }
+
     pub fn federation_id_prefix(&self) -> FederationIdPrefix {
         self.0
             .iter()
             .find_map(|data| match data {
                 OOBNotesData::FederationIdPrefix(prefix) => Some(*prefix),
+                OOBNotesData::Invite { federation_id, .. } => Some(federation_id.to_prefix()),
```

Comment:
@dpc feels like this is what you were asking for in your comment?

Reply:
:+1: